torch/_higher_order_ops/associative_scan.py

```diff
@@ -141,12 +261,6 @@ def trace_associative_scan(
             f"combine_fn output type mismatch, expected {i.dtype} "
             + f"but got {o_meta.dtype}"
         )
-        assert (
```

Comment:
why remove the check here?

Reply:
This check was initially introduce by Peter to ensure that only scalar shapes are returned. However, this constraint is relaxed now and the shapes my be non-scalar. The same check was relaxed in [_higher_order_ops.py](https://github.com/pytorch/pytorch/pull/133012/files#diff-d6c3a57794a9a8a79eacedc57d68ff631fe80776f8701398eb3ef349973b033bL1072)

Reply:
Is the restriction relaxed because we added generic combine mode? Yeah, we should remove the output shape check in dynamo. But trace_associative_scan will only be triggered when combine_mode = "pointwise", right? So we're essentially removing the constraint for pointwise, is this what we intended?

Reply:
No, the restriction is not necessarily relaxed because of `combine_mode='generic'`. I don't think that the restriction was initially there because of any failure, but rather because scalar functions are inherently pointwise. However, there are more pointwise functions that are not scalar and after a discussion with Peter, we removed the check.